File: BoolValue.hs
{-# LANGUAGE MultiParamTypeClasses, TypeFamilies, DataKinds #-}

module BoolValue where

import Java
import ExpValue

data BoolValue = BoolValue @com.eunmin.BoolValue deriving Class

foreign import java unsafe "@new" newBoolValue :: Bool -> Java a BoolValue

foreign import java unsafe getValue :: Java BoolValue Bool

type instance Inherits BoolValue = '[ExpValue]

File: java/com/eunmin/BoolValue.java
package com.eunmin;

public class BoolValue implements ExpValue {
  private Boolean value;

  public BoolValue(boolean value) {
      this.value = value;
  }

  public boolean getValue() {
    return value;
  }

  @Override
  public String getName() {
    return "BoolValue";
  }
}
